src/core_modules/capture-core/components/TrackedEntityTypeSelector/TrackedEntityTypeSelector.component.js

```diff
@@ -60,11 +63,14 @@ const InfoOutlinedIconWithStyles = withStyles({
 })(InfoOutlinedIcon);
 
 export const TrackedEntityTypeSelectorPlain =
-  ({ classes, onSelect, selectedSearchScopeId }: Props) => {
+  ({ classes, onSelect }: Props) => {
       const trackedEntityTypesWithCorrelatedPrograms = useTrackedEntityTypesWithCorrelatedPrograms();
+      const dispatch = useDispatch();
+      const selectedSearchScopeId = useCurrentTrackedEntityTypeId();
 
       const handleSelectionChange = ({ selected }) => {
           onSelect(selected, scopeTypes.TRACKED_ENTITY_TYPE);
+          dispatch(setTrackedEntityTypeIdOnUrl({ trackedEntityTypeId: selected }));
```

Comment:
Not a big fan of putting this directly in the component file here. I have suggestion and you can tell my why that would not work: Remove the dispatch from here. Add a callback method in `NewPage.container` and call that from from https://github.com/dhis2/capture-app/blob/1b119d401fc52bec01368b5b43b98303bf7aefc2/src/core_modules/capture-core/components/Pages/New/NewPage.component.js#L74. For the `selectedSearchScopeId` on line 69 in this file, I would prefer receiving this from the parent component as you did earlier. 

Could be that I'm missing some reason why this would not work so please feel free to enlighten me 🙂

Reply:
> Not a big fan of putting this directly in the component file here. I have suggestion and you can tell my why that would not work: Remove the dispatch from here. Add a callback method in NewPage.container and call that from from

I agree with what you say that this could be in a container. We had discussed before that we would like to keep separate the components from the containers, something I obviously dont do here. I will change that. 

One thing though. I think its more fitting for the `dispatch(setTrackedEntityTypeIdOnUrl({ trackedEntityTypeId: selected }));` to be part of the `TrackedEntityTypeSelector` module. As far as I can see we will always want the same redux action to be triggered here. Aka the `setTrackedEntityTypeIdOnUrl` action. 

So now you are right, this could also happen in the `NewPage.container` and be passed here as prop. However that indicates that we would like to to pass in different actions than the `setTrackedEntityTypeIdOnUrl` in the `TrackedEntityTypeSelector` component. I cant think of a use case for it. At least not with the current use of the `TrackedEntityTypeSelector`. We now use it in the search page and in the new page and in both cases we want the action to change the url and the action can be in both places the same. I would leave it as it is for now and if we find ourselves confronted with cases that the component needs to trigger extra actions then I would "open" it.

--- 

Now about the line 69 @JoakimSM in which file is it again?

Reply:
Aha, I didn't notice this file was reused. Alright, lets leave it as is for now then (apart from the container part).

Reply:
yes you right I will change the container part.

>  For the selectedSearchScopeId on line 69 in this file, I would prefer receiving this from the parent component as you did earlier.

Can you help me with that? Cant really find which `selectedSearchScopeId` you referring to

Reply:
Never mind, let's leave it as is 👍